File: app/src/main/java/ru/p3tr0vich/fuel/utils/UtilsString.java
package ru.p3tr0vich.fuel.utils;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.TextUtils;

public class UtilsString {

    private UtilsString() {
    }

    @NonNull
    public static String encodeLineBreaks(@Nullable String s) {
        if (TextUtils.isEmpty(s)) return "";

        return s.replaceAll("\\\\n", "\\\\\\\\n").replaceAll("[\\n]", "\\\\n");
    }

    @NonNull
    public static String decodeLineBreaks(@Nullable String s) {
        if (TextUtils.isEmpty(s)) return "";

        return s.replaceAll("(?<![\\\\])\\\\n", "\n").replaceAll("\\\\\\\\n", "\\\\n");
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/utils/UtilsString.kt
package ru.p3tr0vich.fuel.utils

import android.text.TextUtils

object UtilsString {

    @JvmStatic
    fun encodeLineBreaks(s: String?): String {
        return if (TextUtils.isEmpty(s)) {
            ""
        } else {
            s!!.replace("\\\\n".toRegex(), "\\\\\\\\n").replace("[\\n]".toRegex(), "\\\\n")
        }
    }

    @JvmStatic
    fun decodeLineBreaks(s: String?): String {
        return if (TextUtils.isEmpty(s)) {
            ""
        } else {
            s!!.replace("(?<![\\\\])\\\\n".toRegex(), "\n").replace("\\\\\\\\n".toRegex(), "\\\\n")
        }
    }
}